### ISSUE TITLE
Sometimes linked lines get unlinked unexpectely

### ISSUE THREAD

**Issue by Rhialto** [MEGA65/mega65-rom-public#146]
**Test Environment (required)**
You can use [MEGA65INFO](https://files.mega65.org?id=3eeb7d82-8bd7-44b7-82b6-cef29d123b0e) to retrieve this.
 - Platform: MEGA65 r6
 - ROM Release: 920395
 - Core Commit: doesn't matter: it's a rom bug

**Describe the bug**
Sometimes text screen lines that are linked together get unlinked unexpectedly.
I have a game in Basic, which can run in 40 column mode and in 80 column mode. Messages are printed on the top 2 lines, because in 40 column mode it sometimes needs more than 40 characters for a message. Later the message is removed, and another one may appear.
I noticed when playing the game that sometimes when a new 2-line message was printed, it was pushing down the whole of the playing field.

**To Reproduce**
- Type in this test program (or load the attched file):
```
;test006a ==2001==
  100 w$="  ":gosub 160
  110 print"see how it moved down? (bug)
  120 sleep 3
  130 w$="..":gosub 160
  140 print"see how it didn't?
  150 end
  160 print"{clr}some text that is between fourty and eighty characters long
  170 print"{home}{down}{down}some text that will be shifted down,
  180 print"if the top line is wiped with spaces
  190 sleep 1
  200 rem erase the top line
  210 print"{home}";:for i = 1 to 35:print w$;:next:print
  220 sleep 1
  230 print"{home}some more text that fills more than 40 chars at the top
  240 print"{home}{down}{down}{down}{down}{down}{down}{down}
  250 return
```
It is printing (almost) the same things to the screen twice, except the first time it wipes the top lines with spaces and the second time with other characters. Note how the behaviour differs of the line "some text that will be shifted down".
- run this in 40 column mode
- remove the PRINT at the end of line 210 and run the program again.
- this time the bug does not appear. 

**Expected behavior**
Text printed over empty space of a pair of linked lines should just print over the  empty space, and not push the remaining text down.

**Additional context**
So what seems to happen is that if you type (or print) a return on a set of linked lines, but the lines are empty, then they get unlinked. This is unexpected since the Editor versions for PET, VIC-20 and C=64 did not do this.
But given this description I started to get the sneaky suspicion that this was somehow meant to be a feature.
Looking through the ROM code, I found in `system.src` a label `prtcr` which is where a CR (or shift-CR) is printed. It  finds the end of the current line and then sets next line as non-continued.
The C64 editor does not have such code. See https://pagetable.com/c64ref/c64disasm/#E72A where the CR is detected, and https://pagetable.com/c64ref/c64disasm/#E891 where it is executed. This only resets various editor attributes (like at `reset_modes`) and then moves to the next line https://pagetable.com/c64ref/c64disasm/#E87C and possibly scrolling once or twice.

There is an easy test to show some weird results of this.
- go to 40 column mode, clear the screen
- type about 50 characters
- type shift-return (avoid distractions like syntax errors)
- cursor ends up just below the typed text
- move cursor up and type spaces over the characters 
- type shift-return again
- cursor ends up at the start of the line where it already was (!!!!)

**Comment by Rhialto** [comment]
Hm it didn't attach the file. Let's try again.
[test006a.prg.gz](https://github.com/user-attachments/files/16109683/test006a.prg.gz)

I would also like to add that some feature to unlink lines can be useful, but the way it is done gets in the way. It makes the strategy of overprinting parts of a carefully arranged screen much to fragile. As an additional escape function it would be totally useful though.

(I checked the C=128 (with Vice of course) and it seem this "feature" was already present there, at least the test steps at the end behave the same. I still don't like it though)

**Comment by dansanderson** [comment]
Both your test program and the manual test you describe have identical behavior on the C128, C65 ROM, and MEGA65 ROM. I agree that it's not intuitive. That's how I would describe line linking in general, given the lack of visual indicators. But it's difficult to justify changing it if the C128 does it, unless there's a much better idea. I suspect it works this way to support 160-character logical lines on a display that could be either 40 or 80 columns wide.

The general solution for programs that want to use PRINT to update the screen without concern for line pushing is to disable line pushing using the escape sequence ESC R (`PRINT CHR$(27)+"R"`). Use ESC N to re-enable. This allows the PRINT command to write a character to the right-most column without pushing other lines downward. Starting your test program with `PRINT CHR$(27)+"R"` eliminates the unwanted behavior. This feature was added in ROM 920396.

Resolving as will-not-fix, but feel free to re-open if you want to discuss further.

**Comment by Rhialto** [comment]
No need to re-open, I just wanted to report the workaround I used, and some others that came to my mind (for people who may find this later).
- I erased my top 2 lines with non-space characters, but printed in the background colour so that you don't see them. If the next message is short, the line nevertheless remains long and linked.
- Or it could be done with shift-spaces which look the same as spaces but don't count as spaces for pretty much all purposes I can think of.
- Or setting a window of 2 lines at the top with ESC-T and ESC-B would prevent pushing down too. Erasing the text can be even easier by scrolling it out of view instead (instead of overprinting it).

The Plus/4 seems to behave much as the C65: it has ESC-T/B window settings, and it looks like it can link more than 2 lines. 
The 128 doesn't appear to have window setting capability (I tried the same character codes as for the 8032).
The 8032 doesn't ever link lines, not even in windowed mode. The memory locations used to keep track of this have been reused for mostly DOS data.
